### ISSUE TITLE
macOS: Possibility of Infinate Loop In Sizing Functions

### ISSUE THREAD

**Issue by mfisher31** [lv2/pugl#109]
Calling move/resize view functions while inside a configure event from calling code causes an infinite loop ...

```c
[window dispatchCurrentConfiguration];
dispatchCurrentChildViewConfiguration(view);
```

... these get called from `puglSetFrame` and related functions which in turn sends another configure event in the same stroke... and down it goes!

I'm avoiding this with simple bool flag here, and handling later in puglUpdate:
https://gitlab.com/lvtk/pugl/-/commit/3a3ede7f3f2e9571fc59230c36be60cebec02e19

Kind of a scratch pad solution, but the infinite loop is real.

**Comment by drobilla** [comment]
I can see what led you here, but you probably never want to actually do that, it's going to be a nightmare in all kinds of ways, and different kinds of ways of different platforms for extra fun.

It should probably fail gracefully when called in this and other inappropriate contexts, as should a bunch of other things.  I've been working on improving the internal state tracking for this among other things, but there's still a lot of checks needed to make things resilient.

**Comment by mfisher31** [comment]
Good call.  Things like this can easily become a nightmare... especially when just "going for it" and it gets baked in the code base forever.
